pkgs/development/tools/dbt/default.nix

```diff
@@ -0,0 +1,178 @@
+{ lib
+, fetchFromGitHub
+, python3Packages
+, rustPlatform
+}:
+
+let
+  mashumaro = python3Packages.buildPythonPackage rec {
+    pname = "mashumaro";
+    version = "3.3.1"; # dbt wants a specific version
+
+    disabled = python3Packages.pythonOlder "3.6";
+
+    src = python3Packages.fetchPypi {
+      inherit pname version;
+      sha256 = "sha256-mX7QpM5klnuW/2X1yna45eRZpOx6ag9zYloGcASoAck=";
+    };
+
+    propagatedBuildInputs = with python3Packages; [
+      typing-extensions
+      msgpack
+      pyyaml
+      tomli
+      tomli-w
+      pydantic
+      orjson
+      marshmallow
+      cattrs
+    ];
+
+    pythonImportsCheck = [ "mashumaro" ];
+  };
+
+  logbook = python3Packages.buildPythonPackage rec {
+    pname = "Logbook";
+    version = "1.5.3";
+
+    src = python3Packages.fetchPypi {
+      inherit pname version;
+      sha256 = "sha256-ZvRUraD1bq5DBm9gSiIrCYk/mMGtwY3xaXEHYbjzL+g=";
+    };
+
+    # Wants to run some kind of connection to something
+    doCheck = false;
+
+    pythonImportsCheck = [ "logbook" ];
+  };
+
+  minimal-snowplow-tracker = python3Packages.buildPythonPackage rec {
+    pname = "minimal-snowplow-tracker";
+    version = "0.0.2";
+
+    src = python3Packages.fetchPypi {
+      inherit pname version;
+      sha256 = "sha256-rKv3Vy2w5/XL9pg9SV7vVAgfcb45IzDrOq25zLOdqqQ=";
+    };
+
+    propagatedBuildInputs = with python3Packages; [
+      six
+      requests
+    ];
+
+    pythonImportsCheck = [ "snowplow_tracker" ];
+  };
+
+  dbt-extractor = python3Packages.buildPythonPackage rec {
+    pname = "dbt_extractor";
+    version = "0.4.1";
+
+    src = python3Packages.fetchPypi {
+      inherit pname version;
+      sha256 = "sha256-dbHGZWmewPH/zhuj13b3386AIVbyLnCnucjwtNfoD0I=";
+    };
+
+    cargoDeps = rustPlatform.fetchCargoTarball {
+      inherit src;
+      name = "${pname}-${version}";
+      sha256 = "sha256-COuFqMxgzPEldfdP9WTZjUq504bLF+qopZYfDQeWLzM=";
+    };
```

Comment:
Since https://github.com/NixOS/nixpkgs/pull/221716/commits/4329de1264ecb7cca8709dc1984b86beb11bcc4e, we will need to use `importCargoLock` because this project contains a git dependency. It will look something like:

```
cargoDeps = rustPlatform.importCargoLock {
  lockFile = ./Cargo.lock;
  outputHashes = {
    "tree-sitter-jinja2-0.1.0" = "sha256-...";
  };
};
```
where `./Cargo.lock` refers to a checked in Cargo.lock file.